backend/src/services/pki-sync/aws-certificate-manager/aws-certificate-manager-pki-sync-fns.ts

```diff
@@ -0,0 +1,687 @@
+            Certificate: cert,
+            PrivateKey: privateKey,
+            Tags: [
+              {
+                Key: INFISICAL_CERTIFICATE_TAG,
+                Value: key
+              }
+            ]
+          };
+
+          if (certificateChain && certificateChain.trim().length > 0) {
+            importParams.CertificateChain = certificateChain;
+          }
+          if (existingArn) {
+            importParams.CertificateArn = existingArn;
+          }
+
+          const response = await withRateLimitRetry(() => acm.importCertificate(importParams).promise(), {
+            operation: "import-certificate",
+            syncId: pkiSync.id
+          });
+
+          return { key, name, success: true, response };
+        } catch (error) {
+          const errorMessage = error instanceof Error ? error.message : "Unknown error";
+          logger.error(
+            {
+              syncId: pkiSync.id,
+              certificateKey: key,
+              certificateName: name,
+              error,
+              errorMessage
+            },
+            "Failed to import certificate to AWS Certificate Manager"
+          );
+
+          throw new PkiSyncError({
+            message: `Failed to import certificate ${key} to AWS Certificate Manager: ${errorMessage}`,
+            cause: error instanceof Error ? error : new Error(errorMessage),
+            context: {
+              certificateKey: key,
+              certificateName: name,
+              region: destinationConfig.region
+            }
+          });
+        }
+      },
+      { operation: "import-certificates", syncId: pkiSync.id }
+    );
+
+    const results = uploadResults;
+    const failedUploads = results.filter((result) => result.status === "rejected");
+    const successfulUploads = results.filter((result) => result.status === "fulfilled");
+
+    let removedCertificates = 0;
+    let failedRemovals = 0;
+    let removeResults: PromiseSettledResult<{ arn: string; success: boolean; error?: Error }>[] = [];
+
+    if (certificatesToRemove.length > 0) {
+      removeResults = await executeWithConcurrencyLimit(
+        certificatesToRemove,
+        async (certificateArn) => {
+          try {
+            await withRateLimitRetry(() => acm.deleteCertificate({ CertificateArn: certificateArn }).promise(), {
+              operation: "delete-certificate",
+              syncId: pkiSync.id
+            });
+            return { arn: certificateArn, success: true };
+          } catch (error) {
+            logger.error(
+              { error, syncId: pkiSync.id, certificateArn },
+              "Failed to remove expired/removed certificate from AWS Certificate Manager"
+            );
+
+            return {
+              arn: certificateArn,
+              success: false,
+              error: error instanceof Error ? error : new Error("Unknown error")
+            };
+          }
+        },
+        { operation: "remove-certificates", syncId: pkiSync.id }
```

Comment:
Looks like this is the same operation as the one in line 631. Maybe we can create a reusable helper for aws CM cert deletion?